flytekit/tools/ignore.py

```diff
@@ -87,6 +87,30 @@ def _is_ignored(self, path: str) -> bool:
         return self.pm.matches(path)
 
 
+class FlyteIgnore(Ignore):
+    """Uses a .flyteignore file to determine ignored files."""
+
+    def __init__(self, root: Path):
+        super().__init__(root)
+        self.patterns = self._parse()
+
+    def _parse(self) -> List[str]:
+        patterns = []
+        flyteignore = os.path.join(self.root, ".flyteignore")
+        if os.path.isfile(flyteignore):
+            with open(flyteignore, "r") as f:
+                patterns = [l.strip() for l in f.readlines() if l.strip() and not l.startswith("#")]
```

Comment:
we may end up removing the dependency in the future, but while we have it, why not just use the PatternMatcher object that the dockerignore object uses?

Reply:
haha just pushed that! since my test was actually failing